WooCommerce/WooCommerceTests/ViewRelated/Survey/SurveyViewControllerTests.swift

```diff
@@ -118,19 +141,36 @@ private extension SurveyViewControllerTests {
 private extension SurveyViewControllerTests {
     final class FormSubmittedNavigationAction: WKNavigationAction {
 
-        private let httpMethod: String
-        init(httpMethod: String) {
-            self.httpMethod = httpMethod
+        static var emptyAction: FormSubmittedNavigationAction {
+            return FormSubmittedNavigationAction(messageTag: nil)
+        }
+
+        static var nonCompletedAction: FormSubmittedNavigationAction {
+            return FormSubmittedNavigationAction(messageTag: "other")
+        }
+
+        static var completedAction: FormSubmittedNavigationAction {
+            return FormSubmittedNavigationAction(messageTag: "done")
+        }
+
+        private let messageTag: String?
+        private init(messageTag: String?) {
+            self.messageTag = messageTag
         }
 
         override var navigationType: WKNavigationType {
             return .formSubmitted
         }
 
         override var request: URLRequest {
-            var request = URLRequest(url: WooConstants.URLs.inAppFeedback.asURL())
-            request.httpMethod = httpMethod
-            return request
+            var urlComponents = URLComponents(url: WooConstants.URLs.inAppFeedback.asURL(), resolvingAgainstBaseURL: false) ?? URLComponents()
+            if let messageTag = messageTag {
+                let item = URLQueryItem(name: "msg", value: messageTag)
+                urlComponents.queryItems = [item]
+            }
+
+            let finalURL = (try? urlComponents.asURL()) ?? WooConstants.URLs.inAppFeedback.asURL()
```

Comment:
I think it's fine to forcefully unwrap the result here. The fallback value was more confusing for me. 🙂 

```suggestion
            let finalURL = try! urlComponents.asURL()
```